python/lsst/pipe/analysis/colorAnalysis.py

```diff
@@ -521,6 +567,124 @@ def runDataRef(self, patchRefsByFilter, subdir=""):
+                cat = cat[colsToLoadList]
+                if dfLoadColumns is None:
+                    dfLoadColumns = colsToLoadList
+                else:
+                    dfLoadColumns.update(column=colsToLoadList)
+            else:
+                cat = parquetCat.toDataFrame(columns=dfLoadColumns)
+            cat = addElementIdColumn(cat, dataRef.dataId, repoInfo=repoInfo)
+            if dfDataset == "forced_src":  # insert some columns from the ref and meas cats for forced cats
+                if refColsToLoadList is None:
+                    ref = parquetCat.toDataFrame(columns={"dataset": "ref",
+                                                          "filter": dataRef.dataId["filter"]})
+                    refColsToLoadList = [col for col in ref.columns if
+                                         (col.startswith(tuple(self.config.columnsToCopyFromRef))
+                                          and not any(s in col for s in self.config.notInColStrList))]
+                    ref = ref[refColsToLoadList]
+                else:
+                    ref = parquetCat.toDataFrame(columns={"dataset": "ref",
+                                                          "filter": dataRef.dataId["filter"],
+                                                          "column": refColsToLoadList})
+                cat = pd.concat([cat, ref], axis=1)
+                if measColsToLoadList is None:
+                    meas = parquetCat.toDataFrame(columns={"dataset": "meas",
+                                                           "filter": dataRef.dataId["filter"]})
+                    measColsToLoadList = [col for col in meas.columns if
+                                          (col.startswith(tuple(self.config.columnsToCopyFromMeas))
+                                           and not any(s in col for s in self.config.notInColStrList))]
+                    meas = meas[measColsToLoadList]
+                else:
+                    meas = parquetCat.toDataFrame(columns={"dataset": "meas",
+                                                           "filter": dataRef.dataId["filter"],
+                                                           "column": measColsToLoadList})
+                cat = pd.concat([cat, meas], axis=1)
+            cat = calibrateSourceCatalog(cat, self.config.analysis.coaddZp)
+            catList.append(cat)
+
+        if not catList:
+            raise TaskError("No catalogs read: %s" % ([dataRef.dataId for dataRef in dataRefList]))
+        allCats = pd.concat(catList, axis=0)
+        # The object "id" is associated with the dataframe index.  Add a
+        # column that is the id so that it is available for operations on it,
+        # e.g. cat["id"].
+        allCats["id"] = allCats.index
+        return allCats
+
     def readCatalogs(self, patchRefList, dataset, repoInfo):
         """Read in and concatenate catalogs of type dataset in lists of data
```

Comment:
same question-- if this `readCatalogs` is substantially duplicated here from `CoaddAnalysisTask`, that seems uncomfortable.